### PR TITLE
feat(addon-editor): add `Iframe` extension

### PR DESCRIPTION
**Pull request by splincode**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows [Conventional Commits](https://www.conventionalcommits.org/en/)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Refactoring
- [ ] Code style update
- [ ] Build or CI related changes
- [ ] Documentation content changes

## What is the current behavior?

Part of #3429

## What is the new behavior?

https://user-images.githubusercontent.com/12021443/214956599-1d9075d8-9660-47c1-82b5-b651f40758d0.mov

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: projects/addon-editor/extensions/iframe-editor/iframe-editor.component.ts [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    ) {
+        super();
+
+        merge(
+            tuiTypedFromEvent(this.documentRef, 'touchend'),
+            tuiTypedFromEvent(this.documentRef, 'mouseup'),
+        )
+            .pipe(takeUntil(destroy$))
+            .subscribe(() =>
+                this.updateAttributes({width: this.width, height: this.height}),
```
Comment (nsbarsukov) [inline projects/addon-editor/extensions/iframe-editor/iframe-editor.component.ts:73]: There are many code duplications with [ImageEditor](https://github.com/Tinkoff/taiga-ui/blob/main/projects/addon-editor/extensions/image-editor/image-editor.component.ts).

Could we solve it somehow ? Maybe some abstract class to extend (or something else)?

File: projects/addon-editor/extensions/iframe-editor/image-iframe.component.html [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    [style.max-height]="options.maxHeight"
+></iframe>
+
+<span
+    class="t-border t-border_right"
+    (tuiPan)="onHorizontalDrag($event, 1)"
+></span>
+<span
+    class="t-border t-border_left"
+    (tuiPan)="onHorizontalDrag($event, -1)"
```
[the remark this answers is not in the document]
Comment (splincode) [reply projects/addon-editor/extensions/iframe-editor/image-iframe.component.html]: fixed

File: projects/addon-editor/extensions/iframe-editor/iframe-editor.component.ts (hunk 1 above)
Comment (splincode) [reply projects/addon-editor/extensions/iframe-editor/iframe-editor.component.ts]: fixed

**Review by vladimirpotekhin — APPROVED** [review]

File: projects/addon-editor/extensions/iframe-editor/iframe-editor.component.ts [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+    get src(): SafeResourceUrl {
+        return this.sanitizer.bypassSecurityTrustResourceUrl(this.attrs.src ?? '');
+    }
+
+    get width(): number | string | null {
+        return this._width || this.attrs.width || null;
+    }
+
+    get height(): number | string | null {
```
Comment (nsbarsukov) [inline projects/addon-editor/extensions/iframe-editor/iframe-editor.component.ts:33]: maybe we should move getters `attrs`,  `width`, `height` inside `AbstractTuiEditorResizable` ? 🤔

Comment (nsbarsukov) [reply projects/addon-editor/extensions/iframe-editor/iframe-editor.component.ts:33]: (any descendant class class simply override any of them, if there is need)

**Review by nsbarsukov — APPROVED** [review]

File: projects/addon-editor/components/editor-resizable/editor-resizable.component.ts [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+    @Input()
+    height: number | string | null = null;
+
+    @Output()
+    readonly sizeChange = new EventEmitter<readonly [width: number, height: number]>();
+
+    @HostBinding('style.width')
+    get hostWidth(): number | string | null {
+        return tuiIsNumber(this.width) ? `${this.width}px` : this.width;
```
Comment (waterplea) [inline projects/addon-editor/components/editor-resizable/editor-resizable.component.ts:32]: tuiPx maybe? Not sure if we need that function though )

File: projects/addon-editor/extensions/iframe-editor/iframe-editor.component.less [hunk 5]
```diff
@@ -0,0 +1,6 @@
+:host {
+    iframe {
```
Comment (waterplea) [inline projects/addon-editor/extensions/iframe-editor/iframe-editor.component.less:2]: Unnecessary nesting

**Review by waterplea — APPROVED** [review]

File: projects/addon-editor/components/editor-resizable/editor-resizable.component.ts (hunk 4 above)
Comment (splincode) [reply projects/addon-editor/components/editor-resizable/editor-resizable.component.ts]: fixed

File: projects/addon-editor/extensions/iframe-editor/iframe-editor.component.less (hunk 5 above)
Comment (splincode) [reply projects/addon-editor/extensions/iframe-editor/iframe-editor.component.less]: fixed

File: projects/addon-editor/extensions/iframe-editor/iframe-editor.component.ts (hunk 3 above)
Comment (splincode) [reply projects/addon-editor/extensions/iframe-editor/iframe-editor.component.ts]: fixed

**Pull request merged**